### PR TITLE
SDI-54: 🐛 Fix exception handling on passthrough

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/ControllerAdvice.kt [hunk 1]
```diff
@@ -34,10 +34,36 @@ class ControllerAdvice {
...
+          .status(HttpStatus.FORBIDDEN.value())
+          .build()
+      )
+  }
+
   @ExceptionHandler(WebClientResponseException::class)
   fun handleException(e: WebClientResponseException): ResponseEntity<ErrorResponse> {
-    if (e.statusCode.value() == 403) return handleWebClientUnAuthorised(e)
-    if (e.statusCode.value() == 401) return handleWebClientForbidden(e)
+    if (e.statusCode.value() == 401) return handleWebClientUnAuthorised(e)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/ControllerAdvice.kt:65]: these were the wrong way round

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/PrisonApi.java [hunk 2]
```diff
@@ -363,6 +364,9 @@ public void deleteAppointments(final List<Long> appointmentIds) {
                         .queryParam("toDate", "{toDate}")
                         .build(offenderNo, fromDate, toDate))
                 .retrieve()
+                .onStatus(s -> s == NOT_FOUND, response ->
+                        response.bodyToMono(ErrorResponse.class).map(r -> new EntityNotFoundException(r.getDeveloperMessage()))
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/PrisonApi.java:368]: handle the not found from prison api

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/controllers/AppointmentControllerTest.kt [hunk 3]
```diff
@@ -154,7 +154,7 @@ class AppointmentControllerTest : TestController() {
           .content(getCreateAppointmentSpecificationAsJson())
       )
         .andDo(print())
-        .andExpect(status().isUnauthorized)
+        .andExpect(status().isForbidden)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/controllers/AppointmentControllerTest.kt:157]: these were mapped incorrectly in the tests

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**